### PR TITLE
Search function for Maniphest

### PR DESCRIPTION
**Pull request by Alex0424**
Adding a new function: task_search.
This function retrieves information about a project, such as columns, tasks, and additional data.

This PR is a first step to implement:
- #63

Left to do after this PR:
- #65
- #66

### CODE REVIEW

**Comment by holmboe** [comment]
Thanks for the PR! 

If you intend to work with upstream it would make sense to create the issue https://github.com/Alex0424/phabfive/issues/1 in the upstream repository instead. Would you mind to do that? https://github.com/dynamist/phabfive/issues

I will do a review of the code and comment on that in follow-up comments.

File: phabfive/cli.py [hunk 1]
```diff
@@ -330,6 +345,17 @@ def run(cli_args, sub_args):
 
         if cli_args["<command>"] == "maniphest":
             maniphest_app = maniphest.Maniphest()
+            if sub_args["list"]:
+                maniphest_app.alex_list(
```
Comment (holmboe) [inline phabfive/cli.py:349]: `alex_list()` does not seem to be included in this PR.

File: phabfive/cli.py [hunk 2]
```diff
@@ -115,6 +115,21 @@
     phabfive maniphest comment add <ticket_id> <comment> [options]
     phabfive maniphest show <ticket_id> ([--all] | [--pp]) [options]
     phabfive maniphest create <config-file> [--dry-run] [options]
+    phabfive maniphest list workboard <board_id> [options]
+    phabfive maniphest list user <user_id> [options]
```
Comment (holmboe) [inline phabfive/cli.py:119]: `alex_list()` does not seem to be included in this PR.

File: phabfive/maniphest.py [hunk 3]
```diff
@@ -14,14 +17,83 @@
...
 
-
 log = logging.getLogger(__name__)
 
-
 class Maniphest(Phabfive):
     def __init__(self):
         super(Maniphest, self).__init__()
 
+    def alex_search(self, created_after=None, updated_after=None, project=None):
```
Comment (holmboe) [inline phabfive/maniphest.py:26]: Would you mind renaming the function to something more suitable, i.e. that doesn't include your name?

File: phabfive/maniphest.py [hunk 4]
```diff
@@ -14,14 +17,83 @@
...
+        """
+        Search for Phabricator Maniphest tasks with given parameters.
+
+        Parameters
+        ----------
+        created_after (int, optional): Number of days ago the task was created.
+        updated_after (int, optional): Number of days ago the task was updated.
+        project (str, optional): Project name.
```
Comment (holmboe) [inline phabfive/maniphest.py:34]: Since all arguments are optional, what happens if you call `phabfive maniphest search` without any arguments? It should probably avoid returning all tasks from the Phabricator instance and instead tell the user that some kind of narrowing down of the search query should be preferred, or add some kind of `--all-tasks` argument that actually returns everything.

I am thinking as an end-user here. What I would expect from a UX perspective.

**Comment by holmboe** [comment]
The title of this PR should be something more descriptive. An example: "Search function for Maniphest"

File: phabfive/maniphest.py [hunk 5]
```diff
@@ -293,3 +374,18 @@ def recurse_commit_transactions(task_config, parent_task_config):
 
         # Always start with a blank parent
         recurse_commit_transactions(parsed_root_data, None)
+
+def days_to_unix(days):
+    """
+    Convert days into a UNIX timestamp.
+    """
+    if days:
```
Comment (holmboe) [inline phabfive/maniphest.py:382]: This if-logic should be moved out to the calling function. This is also a utility function, so apply the same thinking as `format_timestamp()`.

File: phabfive/maniphest.py [hunk 6]
```diff
@@ -14,14 +17,92 @@
...
+        Search for Phabricator Maniphest tasks with given parameters.
+
+        Parameters
+        ----------
+        created_after (int, optional): Number of days ago the task was created.
+        updated_after (int, optional): Number of days ago the task was updated.
+        project (str, required): Project name.
+        """
+
+        if not project:
```
Comment (holmboe) [inline phabfive/maniphest.py:37]: Since `project` is now not optional anymore we need to change the function signature to reflect this.

1. Non-optional arguments comes before optional ones.
2. The function signature should not have a default (currently `None`) to reflect that this needs to be passed to the function.

File: phabfive/cli.py [hunk 7]
```diff
@@ -115,6 +115,19 @@
     phabfive maniphest comment add <ticket_id> <comment> [options]
     phabfive maniphest show <ticket_id> ([--all] | [--pp]) [options]
     phabfive maniphest create <config-file> [--dry-run] [options]
+    phabfive maniphest search [options]
```
Comment (holmboe) [inline phabfive/cli.py:118]: Non-optional arguments are not passed as a generic "`[options]`" in docopt. You should use the "`<board_id>`" variant. Read the docopt documentation or look at other code examples in this repository.

File: phabfive/maniphest.py (hunk 4 above)
Comment (Alex0424) [reply phabfive/maniphest.py]: --all-tasks argument is not required since the function already returns all tasks for the given project.
